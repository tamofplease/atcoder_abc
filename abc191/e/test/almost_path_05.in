38 998
4 8 42831
38 23 73159
10 2 92344
31 15 23684
18 22 43808
17 16 8873
34 1 20504
28 34 26066
1 30 3289
9 15 50284
13 35 88784
31 22 70666
33 37 11221
38 27 17481
17 1 80160
19 1 11679
36 14 94133
17 6 46978
3 29 23805
20 31 90761
19 26 60509
5 35 98213
14 34 27814
13 14 40884
14 17 34096
9 2 49586
32 33 22993
6 37 89920
16 38 60234
22 21 7219
1 36 89729
20 10 22917
32 17 63168
31 1 2959
25 14 58072
17 18 27739
24 27 14562
36 18 72664
26 6 2316
14 28 67091
6 30 31615
32 35 67932
1 13 36149
33 23 92462
17 13 80643
13 22 7148
15 2 43371
29 14 47638
8 18 13268
7 5 55870
19 10 50025
15 16 27199
24 22 9974
32 21 16747
5 34 80552
21 8 87464
17 21 24398
7 19 78577
3 16 41527
35 38 35811
26 18 5788
31 24 96242
6 5 8876
35 7 64015
16 12 44907
25 29 76311
15 17 75128
26 11 89170
38 15 86741
4 14 8516
33 11 88570
35 9 61474
2 16 38231
21 37 97398
10 11 59317
16 18 91856
4 21 13777
16 35 14210
21 17 12657
9 31 2763
36 6 1785
26 7 96667
36 26 90854
24 3 85439
26 30 94225
19 37 15692
34 6 60075
4 37 17715
33 1 96427
23 1 59718
17 25 49055
31 25 61747
26 14 75259
1 34 40269
11 12 56691
28 22 86577
21 20 44369
22 14 30073
16 13 77738
5 1 64913
14 8 67124
16 25 30095
35 25 84002
25 7 66574
15 13 24809
6 27 36168
37 19 71095
31 26 39075
32 19 35417
14 31 77030
17 26 63869
8 11 87538
24 8 34311
34 7 33994
8 19 72132
32 31 14869
27 24 97216
31 32 39224
28 11 41459
29 18 54952
10 35 35309
38 15 31525
11 29 39369
19 7 41179
35 21 99498
12 16 74285
12 13 12879
17 35 19
10 5 28519
30 25 55720
19 6 36939
27 2 56445
11 16 47305
18 32 23438
32 13 73211
22 36 77766
20 8 2933
31 11 15354
29 11 525
21 34 33186
32 11 88396
8 15 23902
6 25 11366
8 9 87296
22 15 93547
1 35 72581
29 18 76692
22 8 17060
38 28 13743
28 7 89549
29 32 11322
31 38 35496
32 23 22967
24 1 87312
34 7 57194
23 24 94596
5 37 93594
5 17 27000
19 32 63510
34 35 68631
24 30 21145
17 14 42002
11 14 92575
27 31 87451
7 2 50771
33 25 80754
10 18 28284
35 17 63224
14 28 1825
19 38 82925
22 14 90272
16 17 59315
32 23 82584
16 31 42474
26 15 30142
23 5 90396
19 13 47523
15 15 97518
31 29 77502
35 25 10953
1 7 51680
17 26 65722
30 13 34534
30 2 60079
20 26 60472
10 25 99640
31 30 36156
21 22 91310
25 13 46272
34 4 36298
20 35 75248
4 23 80969
38 23 42887
10 37 55047
19 14 9392
3 19 30885
33 15 38257
31 23 88752
17 25 69857
34 23 13281
5 25 34045
16 22 22834
16 26 5683
18 33 99865
8 19 44068
21 36 9917
27 38 29613
10 33 60818
33 34 53069
30 9 31205
26 16 72459
11 16 32535
17 36 78559
28 36 41702
23 11 37494
37 1 57440
22 27 23
37 25 866
21 15 27030
31 28 54126
4 4 92564
12 36 48137
4 10 28775
19 26 80742
10 19 90131
32 27 5702
23 14 14103
23 12 35964
36 28 71635
14 20 57165
30 13 15451
17 21 26004
23 2 52161
34 22 72117
6 10 40791
14 6 86175
21 7 81179
30 31 71639
18 19 34602
38 20 53155
1 2 50220
9 8 90579
5 36 67235
11 37 29154
28 26 80080
29 30 39074
14 34 73718
20 36 57185
34 23 51425
38 38 22607
20 38 31117
36 4 27718
2 24 11826
26 36 99040
8 12 1052
17 31 6235
9 24 53973
18 12 36831
1 14 45654
20 33 70342
8 8 79043
11 38 49392
24 23 8676
19 28 71178
16 8 7106
22 26 62056
16 26 62119
21 4 95684
16 18 83795
28 3 89025
31 21 95625
16 5 63194
1 20 66662
18 15 25052
20 20 84818
33 19 6957
30 16 39167
24 30 58756
9 37 96531
2 27 28094
33 7 57231
29 28 3989
4 9 43147
30 32 25619
24 24 11378
22 37 8930
24 36 3231
29 28 66318
21 1 40743
23 4 27839
34 29 67633
10 11 36410
38 33 59930
7 17 56726
33 23 65096
27 30 64584
19 34 46054
9 33 59787
29 1 90304
21 32 74693
35 23 87773
19 22 79110
19 28 51295
15 11 79480
22 18 36004
26 31 31453
13 15 21016
1 32 67007
36 6 92733
9 21 42887
1 33 50473
38 24 73488
30 29 2286
36 15 76621
6 31 19809
24 16 23662
2 12 40388
32 9 32377
14 36 30452
26 27 68471
31 25 61027
6 12 1852
19 36 3777
29 12 55491
5 24 11861
32 22 8790
32 1 75586
38 16 91456
25 2 87229
36 23 84687
30 13 26655
7 4 48343
17 13 34976
6 13 63491
36 6 24857
1 7 84249
25 26 97077
11 17 28807
10 11 44944
35 1 5367
11 34 43180
21 1 89194
7 15 9601
19 33 15631
11 21 81609
12 32 57113
35 6 27866
11 14 93301
10 9 34146
24 2 15945
15 26 89357
19 28 2745
37 33 38295
22 31 2801
38 18 44485
5 5 54493
11 1 60958
24 33 32549
14 32 14619
10 7 39814
25 13 78934
32 19 98293
4 12 45182
2 22 62374
27 28 69117
20 4 56094
21 2 74021
32 28 2982
37 32 52932
12 13 82339
8 7 90392
35 32 44454
15 1 44079
37 16 21587
1 35 54018
21 8 13668
11 10 97655
4 38 14731
19 14 91664
22 10 51918
16 2 40212
23 35 43172
26 17 92539
5 15 10517
15 1 4537
19 20 31205
23 16 24649
31 22 62409
12 5 69588
3 17 62753
29 30 64676
15 20 87607
11 16 50903
20 9 37601
27 1 72244
3 36 60198
5 29 94067
18 14 2673
5 8 25847
8 32 48391
38 34 57439
26 5 12711
21 34 85513
1 14 29494
35 23 79917
1 16 76056
29 3 14867
19 16 70946
13 31 10521
30 4 15576
34 27 57681
36 37 27977
7 10 68369
3 38 84768
10 1 36966
13 25 32125
28 18 16649
29 38 67267
12 34 14088
28 15 68378
24 35 19563
6 1 79160
35 36 26752
33 21 92647
12 30 75193
22 36 63064
26 18 92479
1 8 26511
31 34 13373
15 1 65771
4 2 74760
34 4 85915
21 13 82991
12 30 2431
33 32 59347
21 19 71515
37 17 89003
33 11 6125
18 5 24765
9 16 11204
23 14 98681
22 6 39926
34 18 17191
8 3 23171
22 28 15718
31 27 54722
10 5 12928
8 15 41209
15 24 68232
8 32 73005
18 33 3216
6 12 88125
28 4 36338
3 33 80033
22 30 13868
32 21 99325
33 36 9218
14 26 56745
16 35 40135
33 22 45979
15 29 693
31 29 46799
34 21 6772
29 19 20844
11 14 58873
34 23 52626
20 2 63096
24 12 92992
27 14 44572
33 4 32579
35 27 26840
4 30 43369
7 7 37919
38 36 73397
19 38 60785
11 1 46490
35 36 15871
13 33 5626
33 3 4355
7 24 14342
13 19 64054
3 35 82841
31 30 33019
37 28 46966
3 4 8648
14 35 67567
12 37 82228
12 4 64577
24 25 19207
12 30 1865
24 14 77556
24 8 7579
14 10 38967
10 27 88153
17 27 61429
18 38 96634
16 12 62783
27 23 31706
13 36 72818
36 2 28358
10 20 4336
29 23 74817
36 34 12456
2 30 10211
21 30 99774
3 15 98781
22 33 54565
29 36 13127
22 20 53926
14 10 63143
18 37 39108
12 27 52710
23 8 97676
9 26 17226
36 6 88805
23 36 5381
4 21 80413
16 25 48921
36 9 20454
26 15 20182
29 17 98421
31 23 13981
6 18 11436
31 16 23107
19 21 23221
2 6 37626
35 37 80595
32 25 88464
22 36 95016
4 3 61293
17 24 42383
22 29 45822
22 16 6189
9 6 591
20 29 64286
30 10 79233
19 10 29170
17 36 97593
30 22 90827
30 21 95585
10 25 49268
28 11 28881
1 4 33514
18 7 76611
12 27 93827
23 35 38443
18 19 7464
25 32 85785
28 29 93348
30 37 36835
8 22 44763
32 13 37783
21 19 75787
8 38 47303
4 16 60653
20 21 65272
34 10 49287
22 20 105
24 14 43785
18 28 73243
24 15 32292
30 25 39442
12 3 12617
7 20 88
24 31 39282
27 5 5377
13 7 85946
6 7 27685
36 30 29297
31 26 81279
33 35 52915
2 16 18642
33 16 88950
23 25 6611
23 20 74145
32 7 96167
36 25 86800
5 9 32206
25 1 67315
4 21 61008
6 3 29825
5 36 95740
36 21 75800
17 34 60108
36 35 47380
35 2 34026
33 28 68143
35 4 16212
21 22 28977
15 9 24202
5 11 78973
31 20 90182
29 5 63523
8 16 78870
6 12 60695
34 12 90153
32 33 21484
16 11 53836
31 10 56596
3 19 56014
24 18 56869
3 3 79846
35 34 72572
19 24 47447
9 8 76485
5 37 28358
31 27 72066
37 16 24713
37 25 53163
5 38 28059
18 18 74872
16 37 19764
36 28 64157
23 28 23370
37 11 35819
20 32 38993
7 30 13065
12 2 82160
4 34 96716
3 31 19365
7 29 58832
21 38 25955
24 6 4812
31 31 54632
14 14 30778
7 5 39280
38 28 96493
31 19 78006
11 6 54788
21 34 54982
6 36 59939
30 4 75322
37 38 20785
21 26 33010
33 33 19249
18 16 90915
31 26 94979
38 20 46527
26 35 15992
18 27 89668
16 30 61604
23 15 1399
2 34 9675
33 36 79929
12 36 89461
37 21 72503
2 23 27736
32 30 53415
11 20 30784
17 2 37467
36 24 52171
23 31 48758
12 26 7873
37 14 45026
6 30 36737
10 6 51109
32 21 22336
30 22 50971
23 11 11115
17 21 69999
25 19 28050
33 3 21862
1 27 17005
29 21 99662
16 15 57158
33 5 61105
24 38 47727
13 36 12261
25 28 80617
18 37 9535
7 10 14331
37 10 50272
33 34 78990
21 36 74411
12 28 30537
12 13 45629
38 25 65975
32 18 78358
6 21 71832
7 11 24570
30 21 2319
35 32 3188
15 32 65591
21 1 71074
15 12 10001
15 31 73995
36 24 17687
6 19 84395
23 9 64552
5 32 97666
1 6 38553
6 29 21619
7 3 88976
14 17 35007
26 36 75322
27 17 44270
23 24 70944
5 6 87177
4 7 83085
4 30 82141
21 17 77246
30 12 6494
20 37 14056
12 23 62630
36 8 62458
37 10 99555
28 17 93073
27 34 44751
38 23 97506
8 6 80943
9 24 91223
16 5 67144
16 36 39123
36 17 12751
35 9 47762
25 34 78809
6 9 25137
14 28 40372
4 19 27953
29 1 48512
38 16 51003
26 3 23036
38 30 65975
11 23 92983
23 23 81105
35 28 24957
3 38 32757
6 16 25998
6 2 2216
4 14 5613
31 3 76627
34 7 93453
33 16 53095
14 25 61744
10 9 45373
33 28 55952
8 26 1768
9 10 98687
21 30 48470
22 22 39312
5 29 19207
22 2 13860
8 12 91570
27 15 8551
29 1 63173
15 35 4024
7 3 91983
6 26 41130
10 21 95653
12 9 99495
13 29 48663
14 15 90810
4 19 60733
8 26 47204
38 9 47858
33 13 69961
25 26 33967
2 37 98695
18 20 9295
36 16 7255
3 11 29123
21 9 26798
18 24 62515
5 22 64263
20 18 98187
31 5 34328
12 8 14465
13 17 121
12 15 48717
30 18 68813
35 18 6364
33 35 10799
7 25 72384
14 2 79767
24 25 40841
37 38 14944
13 4 18036
22 27 33966
21 8 28157
36 6 81
2 30 57151
4 27 89866
27 16 95648
24 18 56535
8 37 19299
18 16 6309
32 12 81805
3 16 18245
12 9 45046
16 1 19570
36 10 46425
19 10 99351
23 31 60760
20 3 49709
18 21 54222
5 18 43426
33 16 57061
17 19 40717
17 6 78451
37 22 40925
2 11 426
2 33 81081
15 32 96436
9 18 66288
24 1 85561
8 6 12000
7 34 95244
23 29 77653
27 15 486
25 38 70623
19 15 86724
33 36 85469
19 4 71992
25 30 86924
19 17 86725
29 29 16514
34 14 70733
15 22 65712
3 35 33252
19 18 497
32 17 3523
30 13 21214
33 20 80915
7 5 94640
1 25 59309
7 25 76621
12 25 34981
17 34 82224
28 29 58923
32 33 5547
17 28 22044
16 7 31124
5 6 97443
26 25 64614
1 35 31608
8 26 7968
6 3 9784
6 21 52926
24 38 36339
18 3 90038
34 27 12844
32 28 63536
2 2 50542
36 28 33873
20 5 2914
4 14 30810
16 38 55372
20 36 82608
14 10 94235
24 12 53213
32 24 39195
22 23 66243
24 13 13922
6 2 77973
9 38 73980
36 35 88705
21 12 24893
10 24 13022
19 2 85581
25 18 3317
23 26 60784
3 23 79046
21 19 17057
17 28 82357
35 25 89221
35 18 64697
37 37 70621
31 23 55835
13 33 18386
33 17 35458
13 35 22440
8 22 21183
15 23 16449
8 29 41897
32 5 84144
10 6 22491
2 31 43657
4 5 43720
16 4 43330
13 15 32588
36 30 46156
31 24 89007
28 34 54922
27 23 70896
29 1 26058
8 11 5402
9 1 79378
5 26 97617
27 9 55980
23 34 98060
34 38 52431
4 10 39863
12 19 98072
10 37 76553
14 20 18629
1 34 68214
26 20 64098
2 36 49540
12 16 61182
14 25 76587
26 6 78239
22 16 80876
3 12 90238
29 34 95962
5 23 69150
17 29 60652
26 15 11887
24 30 69115
23 38 22901
21 18 85520
2 16 43203
3 37 68773
21 21 90462
11 18 71409
31 9 94491
7 34 57678
4 36 16923
6 5 91859
7 23 55630
5 31 70822
36 12 27611
10 3 67457
11 22 50168
8 10 29698
37 24 39173
12 29 60647
18 17 6297
13 21 96670
15 1 92441
11 13 63722
4 7 44063
17 25 90977
6 25 21057
20 23 32907
17 1 33754
12 10 905
14 12 35178
33 23 43109
7 30 89172
28 2 76477
7 16 64862
38 35 91153
24 6 65198
17 18 24001
9 17 60521
19 5 92809
28 15 78590
25 3 83233
14 6 78893
18 21 39046
17 19 19939
35 34 37379
11 21 41567
36 35 4632
4 25 77597
15 17 2737
13 28 50625
6 22 72655
1 2 69684
11 2 16601
17 14 2675
31 18 2955
16 12 9781
9 10 81930
22 10 70394
11 21 26514
14 10 26194
7 24 83781
15 35 50631
25 28 88767
16 34 41213
36 12 84535
11 28 37010
28 13 59611
32 14 5000
10 21 38353
8 19 93273
23 33 87677
13 24 54913
16 28 78829
7 27 86365
2 3 8600
4 27 74318
5 13 45214
21 21 46630
21 29 42554
30 14 59237
13 4 54983
29 19 52422
10 6 84908
14 31 36326
7 8 48328
28 19 65156
33 22 94981
16 35 62058
9 9 84599
10 33 15198
35 33 53758
