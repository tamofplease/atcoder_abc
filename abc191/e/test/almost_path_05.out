22028
17027
29891
35436
24049
12041
34790
9226
25728
9226
17027
8359
24117
19619
12132
17780
6730
7961
7961
9226
16034
15243
17848
11378
13718
12041
12132
21147
16514
8359
15487
6730
10670
28661
6730
16034
13718
22607
